core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -4486,6 +4858,124 @@ private long readMinTimestamp(long partitionTimestamp) {
+                if (detachedTxReader == null) {
+                    detachedTxReader = new TxReader(ff);
+                }
+                detachedTxReader.ofRO(path, partitionBy);
+                detachedTxReader.unsafeLoadAll();
+
+                try {
+                    path.trimTo(pathLen);
+                    long partitionSize = detachedTxReader.getPartitionSizeByPartitionTimestamp(partitionTimestamp);
+                    if (partitionSize <= 0) {
+                        throw CairoException.instance(0)
+                                .put("partition is not preset in detached txn file [path=")
+                                .put(path).put(", partitionSize=").put(partitionSize).put(']');
+                    }
+
+                    // Read min and max timestamp values from the file
+                    readPartitionMinMax(ff, partitionTimestamp, path.trimTo(pathLen), columnName, partitionSize);
+                    return partitionSize;
+                } finally {
+                    Misc.free(detachedTxReader);
+                }
+            }
+
+            // No txn file found, scan the file to get min, max timestamp
+            // Scan forward while value increases
+
+            dFile(path.trimTo(pathLen), columnName, COLUMN_NAME_TXN_NONE);
+            final long fd = TableUtils.openRO(ff, path, LOG);
+            try {
+                long fileSize = ff.length(fd);
+                if (fileSize <= 0) {
+                    throw CairoException.instance(ff.errno())
+                            .put("timestamp column is too small to attach the partition [path=")
+                            .put(path).put(", fileSize=").put(fileSize).put(']');
+                }
+                long mappedMem = mapRO(ff, fd, fileSize, MemoryTag.MMAP_DEFAULT);
+                try {
+                    long maxTimestamp = partitionTimestamp;
+                    long size = 0L;
+
+                    for (long ptr = mappedMem, hi = mappedMem + fileSize; ptr < hi; ptr += Long.BYTES) {
+                        long ts = Unsafe.getUnsafe().getLong(ptr);
+                        if (ts >= maxTimestamp) {
+                            maxTimestamp = ts;
+                            size++;
+                        } else {
+                            break;
+                        }
+                    }
+                    if (size > 0) {
+                        detachedMinTimestamp = Unsafe.getUnsafe().getLong(mappedMem);
+                        detachedMaxTimestamp = maxTimestamp;
+                    }
+                    return size;
+                } finally {
+                    ff.munmap(mappedMem, fileSize, MemoryTag.MMAP_DEFAULT);
+                }
+            } finally {
+                ff.close(fd);
+            }
+        } finally {
+            path.trimTo(pathLen);
+        }
+    }
+
+    private void rebuildDetachedColumnIndex(long partitionTimestamp, long partitionSize, Path path, CharSequence columnName) {
```

Comment:
same, this is part of "attach" workflow, all relevant methods should be grouped by workflow

Reply:
there are also class members to do with "attach" workflow that are called "detached"

Reply:
renamed to `rebuildAttachedPartitionColumnIndex`